File: mpp-proiect-repository-saboalina-Lab3Java/src/main/java/domain/validators/AngajatValidator.java
package domain.validators;

import domain.Angajat;

public class AngajatValidator implements Validator<Angajat> {
    @Override
    public void validate(Angajat entity) throws ValidationException {
        StringBuilder err = new StringBuilder();
        if (entity.getId().length()==0)
            err.append("id angajat invalid\n");
        if(entity.getNume().length()==0)
            err.append("nume angajat invalid\n");
        if(entity.getPrenume().length()==0)
            err.append("prenume angajat invalid\n");
        if(entity.getParola().length()==0)
            err.append("parola angajat invalida\n");
        if(err.length() > 0)
            throw new ValidationException(err.toString());
    }
}

File: C#/Laborator2C/domain/validators/AngajatValidator.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Lab2C.domain.validators
{
    class AngajatValidator : IValidator<Angajat>
    {
        public void Validate(Angajat e)
        {
            StringBuilder err = new StringBuilder();
            if (e.ID.Length == 0)
                err.Append("id angajat invalid!\n");
            if (e.Nume.Length == 0)
                err.Append("nume angajat invalid\n");
            if (e.Prenume.Length == 0)
                err.Append("prenume angajat invalid\n");
            if (e.Parola.Length == 0)
                err.Append("parola angajat invalid\n");
            if (err.Length > 0)
                throw new ValidationException(err.ToString());

        }
    }
}
